src/cert_utils.erl

```diff
@@ -0,0 +1,86 @@
+-module(cert_utils).
+-author('denys.gonchar@erlang-solutions.com').
+
+-export([ get_cert_domains/1,
+          get_common_name/1,
+          get_xmpp_addresses/1
+        ]).
+
+-include_lib("public_key/include/public_key.hrl").
+-include("XmppAddr.hrl").
+-include("jlib.hrl").
+
+-type certificate() :: #'Certificate'{}.
+
+-spec get_common_name(certificate()) -> [].
+get_common_name(Cert) ->
+  try
+    {rdnSequence, RDNSequence} = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.subject,
+    [{ok, {_, CN}}] = ['OTP-PUB-KEY':decode('X520CommonName', V) ||
+      AtributesList <- RDNSequence,
+      #'AttributeTypeAndValue'{type = ?'id-at-commonName', value = V} <- AtributesList],
+    CN
+  catch
+    _:_ -> error
```

Comment:
report error here, we are loosing stacktrace info.

Reply:
i'm not gonna log here anything, it's normal when certificate is missing some of the fields, generally  it's not an error. if you want to log something - do it in the proper place, it's a duty of the caller

Reply:
It's not about logging but returning a meaningful error, so the caller will actually know what went wrong.